pkgs/development/python-modules/mautrix-telegram/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, python3 }:
+
+let
+
+  localPython = python3.override {
```

Comment:
It's both because an other tool need to import it to work

Reply:
As you see in https://github.com/tulir/mautrix-telegram/commit/a258c59ca3558ad91b1fee190c624763ca835d2f, their pinning of telethon is completely useless. Just apply that patch to make it work.

Reply:
> an other tool need to import it to work

Would you mind adding that tool to this PR or at least giving me a link?

Reply:
https://github.com/tulir/mautrix-telegram/wiki/Bridge-setup

It's the `alembic upgrade head` step that is importing mautrix-telegram

Reply:
> 
> 
> As you see in [tulir/mautrix-telegram@a258c59](https://github.com/tulir/mautrix-telegram/commit/a258c59ca3558ad91b1fee190c624763ca835d2f), their pinning of telethon is completely useless. Just apply that patch to make it work.

There must be other patches needed as it will not run with telethon 1.5:
```
ImportError: cannot import name 'UpdatePinnedMessageRequest' from 'telethon.tl.functions.channels'
```

Reply:
I added it in `all-packages.nix` using `toPythonApplication` and made a small patch to fix conflicts

Reply:
That's not what I meant. It should be moved out of pythonPackages so that you can use packageOverrides. If you ever need to use it with alembic you can use `pkgs.python3.withPackages (ps: [ ps.alembic (ps.toPythonModule pkgs.mautrix-telegram) ])`.

Reply:
I applied patches so I don't need packageOverrides anymore, and the manual is suggesting that this is the preferred method when a python library is also a application.

Reply:
> the manual is suggesting that this is the preferred method when a python library is also a application

I'd say it's not actually a library, just a strange requirement of using alembic.
What's your stance @worldofpeace?

Reply:
> > the manual is suggesting that this is the preferred method when a python library is also a application
> 
> I'd say it's not actually a library, just a strange requirement of using alembic.
> What's your stance @worldofpeace?



I agree that this is an application and belongs in `all-packages.nix` and alembic just has this requirement.

Reply:
>  If you ever need to use it with alembic you can use pkgs.python3.withPackages (ps: [ ps.alembic (ps.toPythonModule pkgs.mautrix-telegram) ]).


So they'd have to do that to setup the bridge.

Reply:
I think it's only used for updating the database scheme.

Reply:
I think it should be ok now